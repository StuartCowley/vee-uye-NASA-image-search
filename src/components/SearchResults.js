import React from "react";
import PropTypes from "prop-types";
import "../styles/search-results.css";

const SearchResults = ({ results }) => {
  const images = results?.map((image, index) => (
    <img
      key={index}
      className="search-results-image"
      src={image}
      alt="search results"
    ></img>
  ));
  return results && results.length > 0 ? (
    <>
      <p className="search-results-text">Search Results</p>
      <div className="search-results-wrap">
        <div className="search-results">{images}</div>
      </div>
    </>
  ) : (
    <p className="search-results-text">No results</p>
  );
};

export default SearchResults;

SearchResults.propTypes = {
  results: PropTypes.arrayOf(PropTypes.string).isRequired,
};
