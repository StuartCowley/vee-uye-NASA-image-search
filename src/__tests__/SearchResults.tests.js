import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("Search Results", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<SearchResults />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the text correctly", () => {
    render(<SearchResults />);
    screen.getByText(/Search Results/);
  });
});
