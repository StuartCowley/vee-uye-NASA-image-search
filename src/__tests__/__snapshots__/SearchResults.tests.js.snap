// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results renders correctly 1`] = `
<DocumentFragment>
  <p
    class="search-results-text"
  >
    Search Results
  </p>
  <div
    class="search-results-wrap"
  >
    <div
      class="search-results"
    >
      <img
        alt="search results"
        class="search-results-image"
        src="https://images-assets.nasa.gov/image/PIA08216/PIA08216~thumb.jpg"
      />
      <img
        alt="search results"
        class="search-results-image"
        src="https://images-assets.nasa.gov/image/PIA22081/PIA22081~thumb.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
